book/src/concepts/dispatcher.md

```diff
@@ -0,0 +1,21 @@
+# Dispatcher
+
+## What is a `Dispatcher`?
+
+Dispatchers are the heart of the ECS infrastructure. They are the executors that decides when the `System`s will be executed so that they don't walk over each other.
+
+When a dispatcher is created, it is associated with the systems that it will execute. It then generates an execution plan that respects mutability rules while maximizing parallelism.
+
+## Respecting mutability rules
+
+When a system wants to access a `Storage` of a resource, they can do so either mutably or immutably. This works just like in Rust: either only one system can request something mutably and no other system can access it, or multiple systems can request something but only immutably.
+
+The dispatcher looks at all the `SystemData` in the systems and builds execution stages.
+
+If you want to have the best performance possible, you should prefer immutable over mutable whenever it is possible. (`Read` instead of `Write`, `ReadStorage` instead of `WriteStorage`)
```

Comment:
This may encourage users to use types with interior mutability "for performance".  That would defeat the purpose of the `ECS`.  One of the primary points is that it doesn't use locking.

Maybe add a note like "We also discourage the use of `Mutex` or `RwLock` in your components or resources, `Write` is preferable to interior mutability."

Reply:
I think that then it should explain really quickly why it's still better.
I would have went for something more like "Please however keep in mind that `Write` is still preferable to interior mutability, such as `Mutex` or `RwLock` for example."